sources/api/schnauzer/src/helpers.rs

```diff
@@ -2737,4 +2737,160 @@ mod test_oci_spec {
+            r#"{ "type": "RLIMIT_LOCKS", "hard": 7, "soft": 8 }"#
+        );
+    }
+
+    #[test]
+    fn oci_spec_max_file_size_resource_limits_test() {
+        let json = json!({"max-file-size": {"hard-limit": 8, "soft-limit": 9}});
+        let rendered = oci_spec_resource_limits(&json).unwrap();
+        assert_eq!(
+            rendered,
+            r#"{ "type": "RLIMIT_FSIZE", "hard": 8, "soft": 9 }"#
+        );
+    }
+
+    #[test]
+    fn oci_spec_max_locked_memory_resource_limits_test() {
+        let json = json!({"max-locked-memory": {"hard-limit": 9, "soft-limit": 10}});
+        let rendered = oci_spec_resource_limits(&json).unwrap();
+        assert_eq!(
+            rendered,
+            r#"{ "type": "RLIMIT_MEMLOCK", "hard": 9, "soft": 10 }"#
+        );
+    }
+
+    #[test]
+    fn oci_spec_max_msgqueue_size_resource_limits_test() {
+        let json = json!({"max-msgqueue-size": {"hard-limit": 10, "soft-limit": 11}});
+        let rendered = oci_spec_resource_limits(&json).unwrap();
+        assert_eq!(
+            rendered,
+            r#"{ "type": "RLIMIT_MSGQUEUE", "hard": 10, "soft": 11 }"#
+        );
+    }
+
+    #[test]
+    fn oci_spec_max_nice_priority_resource_limits_test() {
+        let json = json!({"max-nice-priority": {"hard-limit": 11, "soft-limit": 12}});
+        let rendered = oci_spec_resource_limits(&json).unwrap();
+        assert_eq!(
+            rendered,
+            r#"{ "type": "RLIMIT_NICE", "hard": 11, "soft": 12 }"#
+        );
+    }
+
+    #[test]
+    fn oci_spec_max_pending_signals_resource_limits_test() {
+        let json = json!({"max-pending-signals": {"hard-limit": 12, "soft-limit": 13}});
+        let rendered = oci_spec_resource_limits(&json).unwrap();
+        assert_eq!(
+            rendered,
+            r#"{ "type": "RLIMIT_SIGPENDING", "hard": 12, "soft": 13 }"#
+        );
+    }
+
+    #[test]
+    fn oci_spec_max_processes_resource_limits_test() {
+        let json = json!({"max-processes": {"hard-limit": 13, "soft-limit": 14}});
+        let rendered = oci_spec_resource_limits(&json).unwrap();
+        assert_eq!(
+            rendered,
+            r#"{ "type": "RLIMIT_NPROC", "hard": 13, "soft": 14 }"#
+        );
+    }
+
+    #[test]
+    fn oci_spec_max_realtime_priority_resource_limits_test() {
+        let json = json!({"max-realtime-priority": {"hard-limit": 14, "soft-limit": 15}});
+        let rendered = oci_spec_resource_limits(&json).unwrap();
+        assert_eq!(
+            rendered,
+            r#"{ "type": "RLIMIT_RTPRIO", "hard": 14, "soft": 15 }"#
+        );
+    }
+
+    #[test]
+    fn oci_spec_max_realtime_timeout_resource_limits_test() {
+        let json = json!({"max-realtime-timeout": {"hard-limit": 15, "soft-limit": 16}});
+        let rendered = oci_spec_resource_limits(&json).unwrap();
+        assert_eq!(
+            rendered,
+            r#"{ "type": "RLIMIT_RTTIME", "hard": 15, "soft": 16 }"#
+        );
+    }
```

Comment:
nit: these could probably all be combined into one test that tests everything.

Typically when there are multiple test cases, I'd expect to see boundary testing - largest positive and smallest negative numbers, hard less than soft, that sort of thing - in addition to the test that exercises the happy path functionality.

I don't think you need more than two cases - one that checks that everything can be rendered, and one that checks for unlimited handling specifically.